src/main/java/fr/cnes/sonar/plugins/hadolint/check/HadolintSensor.java

```diff
@@ -196,22 +201,41 @@ static void saveIssue(final SensorContext context, final Map<String, InputFile>
      * @return Return a list of path 'findable' in the file system.
      */
     private List<String> getReportFiles(final Configuration config, final FileSystem fileSystem) {
-        // Contains the result to be returned.
-        final List<String> result = new ArrayList<>();
+        // List of found reports
+        List<String> result = new ArrayList<>();
 
         // Retrieves the non-verified path list from the SonarQube property.
-        final String[] pathArray = config.getStringArray(HadolintPluginProperties.REPORT_PATH_KEY);
+        String reportPathPropertyKey = HadolintPluginProperties.REPORT_PATH_KEY;
+        final String reportPaths = config.get(reportPathPropertyKey)
+                .orElse(HadolintPluginProperties.REPORT_PATH_DEFAULT);
 
         // Check if each path is known by the file system and add it to the processable
-        // path list,
-        // otherwise print a warning and ignore this result file.
-        for (String path : pathArray) {
-            final File file = new File(fileSystem.baseDir(), path);
-            if (file.exists() && file.isFile()) {
-                result.add(path);
-                LOGGER.info(String.format("Result file %s has been found and will be processed.", path));
+        // path list, otherwise print a warning and ignore this report file.
+        DirectoryScanner scanner = null;
+        List<String> includedFiles = null;
+        String baseDirPath = fileSystem.baseDir().getPath();
+
+        for (String reportPath : reportPaths.split(",")) {
+            scanner = new DirectoryScanner(new File(baseDirPath), WildcardPattern.create(reportPath));
+            includedFiles = scanner.getIncludedFiles();
+
+            if (includedFiles.isEmpty()) {
+                if (config.hasKey(reportPathPropertyKey)) {
+                    // try absolute path
+                    File file = new File(reportPath);
+                    if (!file.exists() || !file.isFile()) {
+                        LOGGER.warn("No report was found using pattern {}", reportPath);
+                    } else {
+                        result.add(file.getPath());
```

Comment:
I do not understand how we can add reports here if includedFiles is empty ? Is it clear for you ?

Reply:
This is a special case inspired from a SonarQube official plugin : 

1. The includedFiles list contains files matching a given pattern (from the plugin property) and found with a dedicated method. 
2. If the property does not contain a pattern but a simple path (e.g. an absolute one), the matching method won't work. In this case, we try to match a unique file using the property as a path, not a pattern. If something matches, we add it to the list.
3. If even the simple path check fails, we don't add anything and in all cases, we notify this failure.

Is it understandable now ?